rotkehlchen/db/eth2.py

```diff
@@ -810,3 +822,110 @@ def get_validators_block_and_mev_rewards(
             mev_rewards_amounts[validator_index] += FVal(amount_str)
 
         return blocks_rewards_amounts, mev_rewards_amounts
+
+    def redecode_block_production_events(self, block_numbers: list[int] | None = None) -> None:
+        """Reprocess eth block production events from the db.
+        - Resets event type depending on whether the fee recipient address is tracked.
+        - Combine mev reward events with evm tx events
+        Optionally limit to only events for the specified block numbers.
+        """
+        with self.db.conn.write_ctx() as write_cursor:
+            for event_type, operation in (
+                (HistoryEventType.STAKING, 'IN'),
+                (HistoryEventType.INFORMATIONAL, 'NOT IN'),
+            ):
+                query = (
+                    'UPDATE history_events SET type=? '
+                    f'WHERE entry_type=? AND subtype=? AND location_label {operation} '
+                    '(SELECT account FROM blockchain_accounts WHERE blockchain=?)'
```

Comment:
This selection happens two times. Would it be probabl better to just query it and put it in a python list and use it in both queries?